### PR TITLE
CAMEL-14809: Clear icon in the search box in navbar to clear th…

### PR DESCRIPTION
**Pull request by psingh587**
- **I have added a Clear/Cancel Button in the search bar to clear the inner contents. It works like this :-**

ezgif com-video-to-gif (2)

- **For now, I have linked a google image as the cancel button. Just wanted to check if this is fine.**

 What more improvements could be done in this?

### CODE REVIEW

File: antora-ui-camel/src/css/header.css [hunk 1]
```diff
@@ -425,6 +425,15 @@ body {
   }
 }
 
+#search-cancel {
+  position: relative;
+  bottom: 39.9%;
+  left: 90%;
```
Comment (zregvart) [inline antora-ui-camel/src/css/header.css:431]: Could we use `rem` in these? Perhaps `padding-left` and `padding-top` instead (haven't tried it).

**Comment by psingh587** [comment]
> Security policy we have in place will not allow that icon to be loaded, and the [icon used](https://www.svgrepo.com/svg/76253/cancel-button) needs attribution. This is very much intentional, we can't be relying on content not served by us. With the exception of Algolia and GitHub at this moment, look at:
> 
> https://github.com/apache/camel-website/blob/60537f0c4d18dce038d2922a72644cc9342cf6ef/static/.htaccess#L1329
> 
> I suggest that you either find a CC0 icon or create one yourself. Make sure the icon is optimized.

Yes @zregvart . 
I will use another icon surely. I used this one to show you the prototype of what i am making. I will get back on this.

**Comment by AemieJ** [comment]
Just a suggestion, on codepen you could search how to create a cancel/close icon using pure css itself. It's quite effective and neat.

**Comment by psingh587** [comment]
> Just a suggestion, on codepen you could search how to create a cancel/close icon using pure css itself. It's quite effective and neat.

Yes, I have already started working upon that.

**Comment by psingh587** [comment]
@zregvart 
Made the required changes. 
- Added a new svg icon.
- Used rem for positioning.

File: layouts/partials/header.html [hunk 2]
```diff
@@ -55,6 +55,7 @@
                 <div class="navbar-fill"></div>
                 <div class="navbar-search results-hidden">
                     <input id="search" class="search" placeholder="Search" autocomplete="off">
+                   <img src= "/_/img/cancel.svg" id="search-cancel">
```
Comment (zregvart) [inline layouts/partials/header.html:58]: Nitpicking
```suggestion
                    <img src= "/_/img/cancel.svg" id="search-cancel">
```

File: antora-ui-camel/src/js/vendor/alogliasearch.bundle.js [hunk 3]
```diff
@@ -93,5 +96,11 @@
         results.innerHTML = ''
       }
     }), 150)
+    cancel.addEventListener('click', function () {
+      container.className = 'navbar-search results-hidden'
+      results.innerHTML = ''
+      search.value = ''
+      cancel.style.display = 'none'
```
Comment (zregvart) [inline antora-ui-camel/src/js/vendor/alogliasearch.bundle.js:103]: Nitpicking: this and the block from 30-32 could be pulled in a separate function.

**Pull request merged**